coderd/metrics.go

```diff
@@ -0,0 +1,34 @@
+package coderd
+
+import "github.com/prometheus/client_golang/prometheus"
+
+type prometheusMetrics struct {
+	workspaceTrafficRx *prometheus.CounterVec
+	workspaceTrafficTx *prometheus.CounterVec
+}
+
+func newPrometheusMetrics(registerer prometheus.Registerer) *prometheusMetrics {
+	workspaceTrafficRx := prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: "coderd",
+			Subsystem: "wireguard",
+			Name:      "workspace_rx_bytes",
+			Help:      "Number of received bytes over Wireguard per workspace.",
+		}, []string{"workspace_id", "workspace_name"},
```

Comment:
Does this become n^2 metrics or just 2n in prometheus?

Reply:
2n, since the same id will always have the same name. I'm not sure what's the best middle ground for adding human readable names or not.